### PR TITLE
Update ccid descriptor dumping to V1.1 spec

### PR DESCRIPTION
**Pull request by rohoog**
Update the dump_ccid_device function to V1.1 spec (2005) and get rid of the warning about only being accurate for V1.0 for most CCID readers.

### CODE REVIEW

File: lsusb.c [hunk 1]
```diff
@@ -2200,29 +2200,34 @@ static void dump_ccid_device(const unsigned char *buf)
 		fputs("          Auto clock change\n", stdout);
 	if ((us & 0x0020))
 		fputs("          Auto baud rate change\n", stdout);
-	if ((us & 0x0040))
+	if ((us & (0x0040 | 0x0080)) == 0x0040)
 		fputs("          Auto parameter negotiation made by CCID\n", stdout);
-	else if ((us & 0x0080))
+	else if ((us & (0x0040 | 0x0080)) == 0x0080)
```
Comment (gregkh) [inline lsusb.c:2205]: This check happens on the if line below again, so this isn't correct anymore.

File: lsusb.c [hunk 2]
```diff
@@ -2130,8 +2130,8 @@ static void dump_ccid_device(const unsigned char *buf)
 	       "        bDescriptorType     %5u\n"
 	       "        bcdCCID             %2x.%02x",
 	       buf[0], buf[1], buf[3], buf[2]);
-	if (buf[3] != 1 || buf[2] != 0)
-		fputs("  (Warning: Only accurate for version 1.0)", stdout);
+	if (buf[3] != 1 || ( buf[2] != 0 && buf[2] != 1))
```
Comment (gregkh) [inline lsusb.c]: Extra ' ' before "buf[2]", please remove.

File: lsusb.c [hunk 3]
```diff
@@ -2200,29 +2200,34 @@ static void dump_ccid_device(const unsigned char *buf)
...
+	else if ((us & (0x0040 | 0x0080)) == 0x0080)
 		fputs("          Auto PPS made by CCID\n", stdout);
 	else if ((us & (0x0040 | 0x0080)))
 		fputs("        WARNING: conflicting negotiation features\n", stdout);
 
 	if ((us & 0x0100))
 		fputs("          CCID can set ICC in clock stop mode\n", stdout);
 	if ((us & 0x0200))
-		fputs("          NAD value other than 0x00 accepted\n", stdout);
+		fputs("          NAD value other than 0x00 accepted (T=1)\n", stdout);
```
Comment (gregkh) [inline lsusb.c:2213]: Please explain somewhere what "T=1" means.

File: lsusb.c [hunk 4]
```diff
@@ -2200,29 +2200,34 @@ static void dump_ccid_device(const unsigned char *buf)
...
-	if ((us & 0x00010000))
+	if ((us & 0x00070000) == 0)
+		fputs("          Character level exchange\n", stdout);
+	else if ((us & 0x00070000) == 0x00010000)
 		fputs("          TPDU level exchange\n", stdout);
-	else if ((us & 0x00020000))
+	else if ((us & 0x00070000) == 0x00020000)
 		fputs("          Short APDU level exchange\n", stdout);
-	else if ((us & 0x00040000))
+	else if ((us & 0x00070000) == 0x00040000)
```
Comment (gregkh) [inline lsusb.c:2223]: again, look at the next if statment, I think it's not correct here anymore, or something needs to be done.

**Review by gregkh — CHANGES_REQUESTED** [review]
Please provide a changelog text that describes what is happening here, it's not obvious at all :(

**Review by gregkh — CHANGES_REQUESTED** [review]
Shouldn't this be part of the previous commit?

File: lsusb.c (hunk 3 above)
Comment (rohoog) [reply lsusb.c:2213]: See iso7816-3. There are 2 standardized options for the low level communication protocol, one character-based protocol (T=0) and one block-based protocol (T=1). The NAD (node-address) communication parameter is only relevant for T=1. People who are familiar with NAD are also familiar with T=1.

File: lsusb.c (hunk 1 above)
Comment (rohoog) [reply lsusb.c:2205]: Why do you say that? The expression 'us & (0x40 | 0x80)' (is the same as us & 0xc0) could have 4 values: 0x00, 0x40, 0x80 or 0xc0. the first if checks for 0x40, the second for 0x80 and the last one for 0xc0, which is an inconsistent value with conflicting flags, hence the warning.

File: lsusb.c (hunk 4 above)
Comment (rohoog) [reply lsusb.c:2223]: Again similar as above, this construction checks that only at most one of the bits in the mask 0x70000 result is set. Multiple bits is a violation of the USB-CCID spec.

File: lsusb.c (hunk 2 above)
Comment (rohoog) [reply lsusb.c]: OK, done

File: lsusb.c (hunk 1 above)
Comment (gregkh) [reply lsusb.c:2205]: Ah, my fault, yes, you are correct.

File: lsusb.c (hunk 4 above)
Comment (gregkh) [reply lsusb.c:2223]: Agreed

File: lsusb.c (hunk 3 above)
Comment (gregkh) [reply lsusb.c:2213]: thanks!

**Review by gregkh — APPROVED** [review]

**Comment by gregkh** [comment]
Thanks for the fixup!

**Pull request merged**